src/test/scala/mesosphere/marathon/state/RootGroupTest.scala

```diff
@@ -31,6 +31,18 @@ class RootGroupTest extends UnitTest with GroupCreation {
       current.app(path) should be('defined)
     }
 
+    "find an app without a parent" in {
+      Given("an existing root group with an app without a parent")
+      val app = AppDefinition("app".toPath, cmd = Some("sleep"))
```

Comment:
This came out of a different test case. I think apps without a parent should be considered invalid.